### PR TITLE
Design mode label

### PR DESCRIPTION
**Pull request by Bjvanminnen**
labelsize

Have label auto-resize as we add text. Provide lock icons that allow you to prevent this from happening.

Couple of open questions for @sfilman and others:
(1) Right now lock icons only prevent autoresizing. Should they prevent other forms of resizing (including entering values into the input box or dragging the resizable in the play area)? One option would be to have the input box disabled when you click lock.
(2) Right now we don't persist lock state anywhere. This means if I run/reset or reload, I return to unlocked. Do we want this to persist?

I tend to be inclined to leave things as is for now.

### CODE REVIEW

File: apps/src/applab/designElements/PropertyRow.jsx [hunk 1]
```diff
@@ -7,8 +12,10 @@ var PropertyRow = React.createClass({
       React.PropTypes.number
     ]).isRequired,
     isNumber: React.PropTypes.bool,
+    lockState: React.PropTypes.string,
```
Comment (joshlory) [inline apps/src/applab/designElements/PropertyRow.jsx]: Could optionally do `React.PropTypes.oneOf([LockState.LOCKED, LockState.UNLOCKED]);`.

**Comment by sfilman** [comment]
Talked to Brent -- plan is to get this checked in and see how it feels. I suspect that we'll want at least to persist the lock state because it seems like it will feel weird without.  Opened https://www.pivotaltracker.com/story/show/96217494 to track

File: apps/src/applab/designMode.js [hunk 2]
```diff
@@ -216,7 +216,18 @@ designMode.onPropertyChange = function(element, name, value) {
...
     default:
-      throw "unknown property name " + name;
+      // Mark as unhandled, but give typeSpecificPropertyChange a chance to
+      // handle it
+      handled = false;
+  }
+
+  if (elementLibrary.typeSpecificPropertyChange(element, name, value)) {
+    designMode.editElementProperties(element);
+    handled = true;
```
Comment (davidsbailey) [inline apps/src/applab/designMode.js]: seems like we should mark this as handled in the case where a typeSpecificPropertyChange handler is found, but no change is made.

**Comment by davidsbailey** [comment]
LGTM

File: apps/src/applab/designMode.js (hunk 2 above)
Comment (Bjvanminnen) [inline apps/src/applab/designMode.js]: Yeah, I guess right now there are two things we care about from child handlers. (1) Does the child know about this type, such that we can consider it a known property and (2) Did the child change something, such that the element properties need to be updated.
Right now those two things are always in sync, but it's not unreasonable to think that they might not be in the future. I don't see a super clean way to handle this (could have a separate function to look to see if the elementLibrary will handle the type, or could have the function return a pair of bools), so I think I'm inclined to leave this be until we need a situation where the two things are not in sync.

**Pull request merged**